app/Models/User.php

```diff
@@ -275,6 +275,24 @@ public function committees()
                 $query->whereNull('committees_users.deleted_at')
                     ->orWhere('committees_users.deleted_at', '>', Carbon::now());
             })
+            ->where('committees.is_society', false)
+            ->where('committees_users.created_at', '<', Carbon::now())
+            ->withPivot(array('id', 'role', 'edition', 'created_at', 'deleted_at'))
+            ->withTimestamps()
+            ->orderBy('pivot_created_at', 'desc');
+    }
+
+    /**
+     * @return mixed Returns all societies a user is currently a member of.
+     */
+    public function societies()
```

Comment:
Is it an idea to abstract functions like these into something like get_committees_or_societies (or get_groups) and do something like return get_groups('committees') and get_groups('societies') for code readability and code re-use?

Reply:
I think these kinds of issues would be better fixed in a next redesign where we really abstract the societies and committees into something like a 'groups' model.

Reply:
I think this should be properly fixed when we actually abstract the model of the societies and committees into for example a "group" model.

Reply:
I see you've marked this as resolved, but don't see any change in this file after the comment has been marked as resolved.

Reply:
Sorry, I thought this was related to the UserProfileController, instead of the User model. I will fix this as soon as possible.